packages/pyright-internal/src/analyzer/semanticTokensWalker.ts

```diff
@@ -131,7 +143,11 @@ export class SemanticTokensWalker extends ParseTreeWalker {
             case TypeCategory.Class:
                 //type annotations handled by visitTypeAnnotation
                 if (!(type.flags & TypeFlags.Instance)) {
-                    this._addItem(node.start, node.length, SemanticTokenTypes.class, []);
+                    const modifiers =
+                        ClassType.isBuiltIn(type) && type.details.moduleName !== 'typing'
```

Comment:
is there a reason we don't want classes from the `typing` module to be marked as `defaultLibrary`? does that match what pylance does?

Reply:
The reason is that I think most people would consider 'builtint' as something that is available without needing to import anything. But according to your comment and the [source code](https://github.com/DetachHead/basedpyright/blob/211d5b08d10d697dc831dfad30d9c180b09116e9/packages/pyright-internal/src/analyzer/typeEvaluator.ts#L16187), pyright considers a lot more things as 'builtin'.

I think it will be better to check the scope of where the class is defined instead of manually excluding modules, the same way I'm checking for functions.

Reply:
This current implementation doesn't account for the aliases within `typing`, i.e. `typing.List`, I would suggest modifying the implementation to:
```ts
ClassType.isBuiltIn(type) && type.aliasName === undefined
```
Or include the `type.details.moduleName === 'builtins'` if there are other things to account for.

Reply:
Actually, if we can just look at the module name here, I don't really see a need for the `isBuiltIn` call.

Reply:
This alias guard also needs to be applied to the other handles

Reply:
Ergh, this would break some of the aliases within `builtins` (`EnvironmentError`)

maybe something like:
```ts
if type.details.moduleName === 'builtins' || type.typeAliasDetails?.moduleName === "builtins"
```

Reply:
> For the aliases, I agree that `List`, `Tuple`, `Dict` and so shouldn't be marked, but `l = list` makes sense to me since is still a reference to the builtin without needing an import.

i agree with this approach, it's consistent with what happens when you assign a function or class to a variable (it inherits its colour)

Reply:
I added @KotlinIsland suggestion and also exclude explicit type aliases (but not regular assignments) from being highlighted as builtins.

Reply:
looks good thanks! if you could add a test as well that'd be great. let me know if you have any issues